db/silkworm/etl/buffer.cpp

```diff
@@ -25,7 +25,13 @@ void Buffer::put(Entry& entry) {
 
 void Buffer::sort() {
     std::sort(entries_.begin(), entries_.end(),
-              [](const Entry& a, const Entry& b) { return a.key.compare(b.key) < 0; });
+              [](const Entry& a, const Entry& b) { 
+                auto diff{a.key.compare(b.key)};
+                if (diff == 0) {
+                    return a.value.compare(b.value) < 0;
+                }
+                return diff < 0; 
+                });
```

Comment:
The returns are not correct imho. They should freely return -1 (for a < b) 0 (for equality) 1 (for a > b)
Here instead you're returning a boolean. I'd go for a much simpler
```suggestion
              [](const Entry& a, const Entry& b) { 
                auto diff{a.key.compare(b.key)};
                return diff ? diff : a.value.compare(b.value);
```

Reply:
MDB_APPENDUP seems to execute without problems so this must right, otherwise  It would throw a key already exist error

Reply:
You're right on this: sorry my fault.
Nevertheless the code could be much simpler by :

```suggestion
              [](const Entry& a, const Entry& b) { 
                return (a.key.compare(b.key) < 0) || (a.value.compare(b.value) < 0);
                }
```

Reply:
Disregard the above ... you're right. I've mixed up. 
std::compare returns three possible values while the compare function must be binary.

Reply:
This is what I meant ... I should go resting.
```suggestion
              [](const Entry& a, const Entry& b) { 
                auto diff{a.key.compare(b.key) || a.value.compare(b.value)};
                return diff < 0; 
                });
```

Reply:
the code you suggested does not work

Reply:
@AndreaLanfranchi IMO the code by @Giulio2002 is fine.
We can also say
```
return a.key < b.key || (a.key == b.key && a.value < b.value);
```
but that might be slower than the original code.

Reply:
N.B. When/if we switch to C++20, its [spaceship operator](https://en.cppreference.com/w/cpp/language/default_comparisons) will hopefully do the right thing out of the box.